cylc/flow/config_schema.py

```diff
@@ -0,0 +1,661 @@
+                'host': [VDR.V_STRING],
+            },
+        },
+        'visualization': {
+            'initial cycle point': [VDR.V_CYCLE_POINT],
+            'final cycle point': [VDR.V_STRING],
+            'number of cycle points': [VDR.V_INTEGER, 3],
+            'collapsed families': [VDR.V_STRING_LIST],
+            'use node color for edges': [VDR.V_BOOLEAN],
+            'use node fillcolor for edges': [VDR.V_BOOLEAN],
+            'use node color for labels': [VDR.V_BOOLEAN],
+            'node penwidth': [VDR.V_INTEGER, 2],
+            'edge penwidth': [VDR.V_INTEGER, 2],
+            'default node attributes': [
+                VDR.V_STRING_LIST, ['style=unfilled', 'shape=ellipse']],
+            'default edge attributes': [VDR.V_STRING_LIST],
+            'node groups': {
+                '__MANY__': [VDR.V_STRING_LIST],
+            },
+            'node attributes': {
+                '__MANY__': [VDR.V_STRING_LIST],
+            },
+        },
+    }
+
+    # Upgrade older suite configurations.
+    u = upgrader(cfg, descr)
+    u.obsolete('6.1.3', ['visualization', 'enable live graph movie'])
+    u.obsolete('6.4.1', ['test battery', 'directives'])
+    u.obsolete('6.11.0', ['state dump rolling archive length'])
+    u.obsolete('7.2.2', ['cylc', 'dummy mode'])
+    u.obsolete('7.2.2', ['cylc', 'simulation mode'])
+    u.obsolete('7.2.2', ['runtime', '__MANY__', 'dummy mode'])
+    u.obsolete('7.2.2', ['runtime', '__MANY__', 'simulation mode'])
+    u.obsolete('7.6.0', ['runtime', '__MANY__', 'enable resurrection'])
+    u.obsolete('7.8.0', ['runtime', '__MANY__', 'suite state polling',
+                         'template'])
+    u.obsolete('7.8.0', ['suite logging', 'roll over at start-up'])
+    u.obsolete('7.8.1', ['cylc', 'events', 'reset timer'])
+    u.obsolete('7.8.1', ['cylc', 'events', 'reset inactivity timer'])
+    u.obsolete('7.8.1', ['runtime', '__MANY__', 'events', 'reset timer'])
+    u.obsolete('7.8.1', ['documentation', 'local index'])
+    u.obsolete('7.8.1', ['documentation', 'files', 'pdf user guide'])
+    u.obsolete(
+        '7.8.1',
+        ['documentation', 'files', 'single-page html user guide']
+    )
+    u.deprecate(
+        '7.8.1',
+        ['documentation', 'files', 'multi-page html user guide'],
+        ['documentation', 'local']
+    )
+    u.deprecate('8.0.0',
+                ['documentation', 'files', 'html index'],
+                ['documentation', 'local']
+                )
+    u.deprecate(
+        '8.0.0',
+        ['documentation', 'urls', 'internet homepage'],
+        ['documentation', 'cylc homepage']
+    )
+    u.obsolete('8.0.0', ['communication'])
+    u.deprecate('8.0.0', ['cylc', 'authentication'], ['cylc', 'authorization'])
+    u.deprecate('8.0.0', ['cylc'], ['general'])
+    u.obsolete('8.0.0', ['enable run directory housekeeping'])
+    u.deprecate(
+        '8.0.0',
+        ['runtime', '__MANY__', 'remote'],
+        ['runtime', '__MANY__', 'job']
```

Comment:
This sort of makes sense - it might be easier to have some sort of switch for a whole section thus:
```
if job.remote_section and job.platform:
    warn deprecated key [remote section]

elif job.remote_section:
    work out which [job platform] to use from [host]
    warn deprecated key [remote section]
    warn how we selected the [job platform]

```

Reply:
```ini
[remote]
    'host'  # transformed into job platform?
    'owner'  # ???
    'suite definition directory'  # move to job
    'retrieve job logs'  # move to job
    'retrieve job logs max size'  # move to job
    'retrieve job logs retry delays'  # move to job
```

Reply:
All the retrieve job logs stuff should platform specific rather than task specific.

Reply:
Looks like I've already added these items to the `[job platforms]` section anyway.